pkg/deploy/elbv2/listener_manager.go

```diff
@@ -365,11 +365,17 @@ func buildSDKMutualAuthenticationConfig(modelMutualAuthenticationCfg *elbv2model
 	if modelMutualAuthenticationCfg == nil {
 		return nil
 	}
-	return &elbv2types.MutualAuthenticationAttributes{
+	attributes := &elbv2types.MutualAuthenticationAttributes{
 		IgnoreClientCertificateExpiry: modelMutualAuthenticationCfg.IgnoreClientCertificateExpiry,
 		Mode:                          awssdk.String(modelMutualAuthenticationCfg.Mode),
 		TrustStoreArn:                 modelMutualAuthenticationCfg.TrustStoreArn,
 	}
+
+	if modelMutualAuthenticationCfg.Mode == string(elbv2model.MutualAuthenticationVerifyMode) {
+		attributes.AdvertiseTrustStoreCaNames = translateAdvertiseCAToEnum(modelMutualAuthenticationCfg.AdvertiseTrustStoreCaNames)
+	}
```

Comment:
Should `IgnoreClientCertificateExpiry` be handled the same way? Or should this be handled the same way as `IgnoreClientCertificateExpiry`?

They are both verify-specific features, this code section is more difficult to understand if similar features are implemented in different ways

Reply:
Unfortunately, the features might be implemented the same way but the API validation is different for the two. When the default value is set for `IgnoreClientCertificateExpiry` the API accepts the input no matter what mode we're using. However, the same is not true for `AdvertiseTrustStoreCaNames`, if the a value is set for this field and the mode is not verify, then the API will fail.

Reply:
Thanks for the clarification! If this is consistent with the API behavior, that's okay with me :)